presto-main/src/main/java/com/facebook/presto/operator/scalar/BitwiseFunctions.java

```diff
@@ -36,7 +36,7 @@ public static long bitCount(@SqlType(StandardTypes.BIGINT) long num, @SqlType(St
         if (bits <= 1 || bits > 64) {
             throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Bits specified in bit_count must be between 2 and 64, got " + bits);
         }
-        long lowBitsMask = (1 << (bits - 1)) - 1; // set the least (bits - 1) bits
+        long lowBitsMask = (1L << (bits - 1)) - 1; // set the least (bits - 1) bits
         if (num > lowBitsMask || num < ~lowBitsMask) {
             throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Number must be representable with the bits specified. " + num + " can not be represented with " + bits + " bits");
         }
```

Comment:
I believe the mask calculation below is also incorrect. It should be ```mask = lowBitsMask | (1L << (bits - 1));```

Reply:
The two logic both work. Because we already checked that input is a valid n-bit number, for positive numbers, all higher bits are 0, for negative numbers all higher bits are 1. So which one do you keep won't affect the result. But I agree 1L << (bits - 1) is logically the correct one.

Reply:
I disagree. It depends whether you consider the top bit the sign bit (because this is the sign bit in `num`) or the `bits`-th bit the sign bit (because this is where the sign bit would be for a `bits`-bit number).

It doesn't really matter. It comes down to which choice is more natural. I find the top bit more intuitive, but some  may find the other choice more intuitive.

Reply:
> the bits-th bit the sign bit (because this is where the sign bit would be for a bits-bit number).

That would be the natural interpretation for an n-bit number, which is what this function computes conceptually. The fact that setting a bit higher than that and using it as a sign bit is a "fluke" of the implementation.

Reply:
This function counts the number of bits with '1', regardless of the signs. So I'd say it's just masking n-bit and count how many 1s are in that n bits. In that sense I agree with @martint more. But another way of thinking about this is to just do `mask = (1L << bits) - 1`. In that way there's no question about which bit to consider as the sign bit. All of these yield the same results so I'm not particularly picky here. But I propose the 3rd solution 😛 (though it's theoretically less efficient than the first 2).

Reply:
All 3 are equivalent.

1. I wrote `long mask = lowBitsMask | 0x8000_0000_0000_0000L; // set the least (bits - 1) bits and the sign bit`
2. @martint suggested `long mask = lowBitsMask | (1L << (bits - 1)); // ...`
3. @rongrong suggested `long mask = (1L << bits) - 1; // ...`

We're probably not going to get a consensus on which mental model is superior. And it's not important.

I'm OK with either 1 or 3. (When you go with 3, remove or change the comment. The comment is describing 1.) I don't like 2 because I feel like what @martint is really advocating for in his reasoning is 3.

@martint Any objections on 3? (@rongrong, change the code to 3 and wait for a pull-request approval from @martint)

Reply:
Yes, 2 and 3 are the same. You either build mask based on lowBitsMask (2) or from scratch (3). I'm fine with 3.

Reply:
I changed to 3 and removed the comment. The logic is pretty clear so I don't think a comment is needed.